### DISCUSSION TITLE
Visualizing State Variable from Abaqus UMAT Interface

### DISCUSSION THREAD

**Discussion by edwardXZ06** [idaholab/moose#18651] [Q&A General] [answered]
I have successfully run a Neo-Hookean umat and a hyperelastic model with Mullin effects umat, both of which work great and save me tons of time developing the relevant material model in C++ separately. Right now, I really like this interface, while I think it can be further improved by adding the feature of visualizing state variables (STATEV). 

Inspecting user-defined state variables is usually more important than reviewing only the total stress/strain in terms of constitutive model development. I am not sure if it is easy to pass STATEV information from materials module to Auxkernals for visualization purpose?

Thank you!

**Comment by dschwen** [comment, marked as answer]
This is super easy using a `MaterialStdVectorAux` AuxKernel. This object allows you to pick a component from the state variable vector property to visualize it.

**Comment by Traiwit** [comment]
Hi @edwardXZ06 and @dschwen are those UMAT available anywhere online, I'm just curious how to implement them into moose?

is it simply just use `AbaqusUMATStress` under `[Materials]`? is there any other section that I need to consider?

```
  [constant]
    type = AbaqusUMATStress
    #                      Young's modulus,  Poisson's Ratio, Yield, Hardening
    constant_properties = '1000 0.3 10 100'
    plugin = ../../plugins/xxxxxx ##(UMAT file)
    num_state_vars = 3
  []
```
Thanks guys!

**Reply by GiudGiud** [reply]
afaik yes

**Comment by BilelELLAFI** [comment]
Hello Mr edrwarXZ06
Could tout please send me the python code of Neo-Hookean umat and a hyperelastic model with Mullin effects umat ?
Thanks
